library/vulnerabilities/ssrf/isRequestToItself.ts

```diff
@@ -0,0 +1,49 @@
+import { getPortFromURL } from "../../helpers/getPortFromURL";
+import { trustProxy } from "../../helpers/trustProxy";
+import { tryParseURL } from "../../helpers/tryParseURL";
+
+// We don't want to block outgoing requests to the same host as the server
+// (often happens that we have a match on headers like `Host`, `Origin`, `Referer`, etc.)
+// We have to check the port as well, because the hostname can be the same but with a different port
+export function isRequestToItself({
+  serverUrl,
+  outboundHostname,
+  outboundPort,
+}: {
+  serverUrl: string;
+  outboundHostname: string;
+  outboundPort: number | undefined;
+}): boolean {
+  // If Node.js is exposed to the internet, we can't be sure about the Host header
```

Comment:
comment is a bit confusing to read, talking about exposed to the internet when using x-forwarded-for and "Host" header only. possible rewrite? The logic is smart though